### DISCUSSION TITLE
6-1 PostGIS Connection on QGIS not working!

### DISCUSSION THREAD

**Discussion by andreasrf** [ua-gist604b-f24/syllabus#27] [General]
QGIS absolutely _refuses_ to connect to localhost so I can finish my assignment. I've tried changing the names around, different SSL modes, adding a new rule to my firewall, using the Host IP instead of the name, trying to connect with the geoserver port (8080), trying to connect to the postgres database, checking my connection settings on postgres, and even deleting the CodeSpace and redoing the whole assignment. I'm finally at my wit's end after giving it a go for a few hours. QGIS version isn't the most recent one

The Error:
> ![image](https://github.com/user-attachments/assets/7f1e1750-38b1-45c3-9948-9febc968a076)

Server Settings:
> ![image](https://github.com/user-attachments/assets/516edcd2-67b7-42e1-b696-296bd1af353c)

QGIS Settings for PostGIS Connection:
> ![image](https://github.com/user-attachments/assets/43a0b022-379e-4174-a525-a87af069abd7)

Thanks!
Andrea

<img src="https://gifdb.com/images/high/pepe-frog-meme-sad-watching-computer-llr03woqoqiuoeii.gif" width="150" height="150">

**Comment by redmesa** [comment]
Hello Andera, 

I spent a few minutes on this today. I was able to reach the PostGIS instance, but I could not successfully connect. I attempted to check if it was an SQL setting in the Docker container, but there were some issues. I’ll try again later.

For the host, you need to use the following format:
glorious-capybara-9vq7vj9rvr29v5v-5432.app.github.dev
not https://glorious-capybara-9vq7vj9rvr29v5v-5432.app.github.dev/
When using this in QGIS, remove the "https://" prefix and the trailing "/".

![image](https://github.com/user-attachments/assets/6d297665-3471-4f5c-9407-c7c85d291f78)


However, this generates the following error:

![image](https://github.com/user-attachments/assets/9df5b10b-4179-4218-af08-365f926f5391)


This error can indicate that the connection can reach the Docker container but cannot establish a connection to it. or a bad user/pw, but I copied and pasted the user/pw. This leads me to believe that the container is not configured to allow external connections, possibly due to issues with port forwarding.

**Reply by andreasrf** [reply]
Agh! Thank you so much for your reply. I have no idea how to solve this since I can't find the right files to edit to allow external connections from the Codespace. Time to email Aaryn!

**Reply by sidneyh-2** [reply]
I am having the same issue but cannot get anything to connect either.

**Reply by mcguckinj** [reply]
Has anyone been able to successfully establish a PostGIS connection in QGIS? I'm stumped on this step.

**Reply by jimmy-nguyen-data-science** [reply]
same I am still stuck here...

**Reply by jenw-ua** [reply]
Nope.

**Reply by aaryno** [reply]
I appreciate everybody trying to resolve this. Unfortunately, I was unable to connect to postgres running in a codespace from QGIS either. Codespaces limits connection types to the port-forwarded codespace ports as HTTP or HTTPS and doesn't support the TCP connection required by postgres. 

For this assignment I will give everybody credit. Nothing more to do here.

If you want to complete this on your own you can run this completely locally. The minimal method for getting this working would be to install VSCode locally and launch it like so, from within your Codespace: 
<img width="327" alt="image" src="https://github.com/user-attachments/assets/294ca9ff-0d3b-48f4-8e99-3ea41c1e4aa8">

This would be an entirely new codespace checked out and running on your computer. It would use your local docker and so postgis would be running locally on your computer. The QGIS connection host would be "127.0.0.1" and the port would be "5432". Everything else should follow from the assignment directions.

If you do this assignment locally I will grant an additional 10 extra credit points for this assignment (20 possible out of 10).

I will follow this up with a class email.

**Reply by travisz09** [reply]
Just FYI, I was able to complete assignment 6.1 in codespaces, just using a slightly different method than what was given. Instead of adding layers to QGIS as a PostGIS layer, I used the WFS method detailed in 6.0. I was able to style my layers, and weirdly the QGIS .sld files worked in 6.1, where they did not work for me in 6.0.

![image](https://github.com/user-attachments/assets/3c9c763e-ea69-4eed-b3ee-2f259c369112)

Thank you @aaryno for helping to clear all this up!
